### ISSUE TITLE
Search path for unit test file doesn't consider current package

### ISSUE THREAD

**Issue by memeplex** [jorgenschaefer/elpy#1690]
In the search path:

```
        (dolist (test (list (format "%s/test_%s" directory file)
                            (format "%s/test/test_%s" directory file)
                            (format "%s/tests/test_%s" directory file)
                            (format "%s/../test/test_%s" directory file)
                            (format "%s/../tests/test_%s" directory file)))
```

the last two lines would probably never match anything because the typical structure is `/tests/package/test_module.py` and not `/tests/test_module.py`.

Why not using test discovery? Or taking the path between the root of the project and the current file to be a package and doing:

```
                            (format "%s/../test/%s/test_%s" directory package file)
                            (format "%s/../tests/%s/test_%s" directory package file)))
```

**Comment by memeplex** [comment]
Also projectile gets it right, maybe you can peep what it's doing.

**Comment by galaunay** [comment]
Projectile is checking every files in the project, which is a very generic solution.
We could do the same:
- if there is a project root, search for an associated test file in the project.
- If not, go one directory up and recursively search for an associated test file (projectile uses `find`, but it is not very windows-friendly...).

The only drawback is that it could be a bit slow.

I would be happy with adding your additional options
``` elisp 
(format "%s/../test/%s/test_%s" directory package file)
(format "%s/../tests/%s/test_%s" directory package file)))
```
It will definitely be faster.

**Comment by galaunay** [comment]
This has been addressed in #1681.
Elpy is now looking for test file in the whole project.
